### PR TITLE
use a .clang-tidy file for the CI action

### PR DESCRIPTION
**Pull request by zingale**
## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
Since this seems to work well, I am going to merge the PR on our fork of the cpp-linter-action:

https://github.com/AMReX-Astro/cpp-linter-action/pull/3

File: .clang-tidy [hunk 1]
```diff
@@ -0,0 +1,37 @@
+Checks: >
+    '-*,
```
Comment (yut23) [inline .clang-tidy:2]: Shouldn't have a single quote here or at the end

File: .clang-tidy [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+       -readability-identifier-length,
+       -readability-implicit-bool-conversion,
+       -readability-isolate-declaration,
+       -readability-magic-numbers,
+       -readability-named-parameter,
+       -readability-simplify-boolean-expr,
+     mpi-*,
+     openmp-*'
+
+HeaderFilterRegex: '[^n][^o][^l][^i][^n][^t]\.H$'
```
Comment (yut23) [inline .clang-tidy:37]: Do note that the `-header-filter` CLI arg overrides this value when run from CI (not sure if that's intentional)

**Review by yut23 — CHANGES_REQUESTED** [review]

File: .clang-tidy (hunk 2 above)
Comment (zingale) [reply .clang-tidy:37]: no, it wasn't intentional.  I just copied the AMReX config

File: .clang-tidy (hunk 1 above)
Comment (zingale) [reply .clang-tidy]: updated

File: .clang-tidy (hunk 2 above)
Comment (zingale) [reply .clang-tidy]: removed

**Review by yut23 — APPROVED** [review]

**Pull request merged**